src/data/nat/prime.lean

```diff
@@ -190,6 +198,15 @@ section min_fac
     ((dvd_prime pp).1 fd).resolve_left (ne_of_gt f2)⟩,
    λ ⟨p2, e⟩, e ▸ min_fac_prime (ne_of_gt p2)⟩
 
+  /--
+  This instance is faster in the virtual machine than `decidable_prime_1`.
+  However it uses `classical.choice`, so can't actually be used in a
+  decision procedure.
```

Comment:
AFAICT, using `classical.choice` is irrelevant. Both instances depend on the same axioms, and if this one used choice in a way that blocked evaluation, it wouldn't compute in the VM either. I think the problem is that this one is implemented using well-founded recursion, which won't reduce (in principle? in practice?) in the kernel.

Reply:
Okay, I think I've corrected the comment, now. Having a second look would be much appreciated. I'm still confused about why we have the `decidable_prime` instance at all, so I know that I'm not writing the right comments.

Reply:
https://xenaproject.wordpress.com/2018/07/26/617-is-prime/

Me being confused about primality being decidable, and Mario's reaction ("make `norm_num` do it")